ignitionrobotics/ign-gazebo
# Update SdfGenerator to save link and sensor data to file

# 🎉 New feature

depends on:
* https://github.com/ignitionrobotics/sdformat/pull/753
* https://github.com/ignitionrobotics/sdformat/pull/757

## Summary

Currently the SdfGenerator only write out models and link poses to SDF file. This PR updates the class to pull more data out from ECM and save them to file. This is needed for saving changes to SDF after models are edited by the model editor 

Specifically, these new SDF elements are now being updated when the world is saved:
* link `<inertial>` and `<wind_mode>` elements
* `<sensor>` element - currently supports only sensor types listed in https://github.com/ignitionrobotics/sdformat/pull/753

## Test it

1. Launch ign-gazebo with `sensors_demo.sdf` or `sensors` world
2. Go to the top left hamburger menu and select `Save world as...`
3. Set the output file by using the `Browse` button or manually input the output sdf file path.
4. Click `OK` to save
5. Open and saved SDF world file and verify content.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge**
